### ISSUE TITLE
[BUG] Playwright Errors swallowed, FFMPEG not found

### ISSUE THREAD

**Issue by NoonRightsWarriorBehindHovering** [microsoft/playwright-java#1280]
### System info
- Playwright Version: 1.33.0
- Operating System: Windows 10 Enterprise
- Browser: [All, Chromium, Firefox, WebKit]
- Other info: Similar issue #782 

### Source code
**Test file (self-contained)**

```java
import com.microsoft.playwright.*;

public class ExampleReproducible {
  public static void main(String[] args) {
    try (Playwright playwright = Playwright.create()) {
      Browser browser = playwright.chromium().launch();
      BrowserContext context = browser.newContext(new Browser.NewContextOptions()
                    .setRecordVideoDir(Paths.get("videos"));
      Page page = context.newPage();
      page.navigate("https://example.com");
      // ...
    }
  }
}
```

**Steps**
- Set Environment Variable PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD to true
- Run above code with and without setRecordVideoDir

**Expected**
Both work or if ffmpeg is not found, show error in logs.

**Actual**
Video Recording fails and exception is swallowed.
Similar error to #782 is shown

```
>       this._firstNonInitialNavigationCommittedReject(new Error('Page closed'));
>                                                     ^
>
> Error: Page closed
>Exception in thread "main" com.microsoft.playwright.PlaywrightException: Playwright connection closed
>	at com.microsoft.playwright.impl.PipeTransport.send(PipeTransport.java:50)
>	at com.microsoft.playwright.impl.Connection.internalSendMessage(Connection.java:164)
>	at com.microsoft.playwright.impl.Connection.sendMessageAsync(Connection.java:130)
>	at com.microsoft.playwright.impl.Connection.sendMessage(Connection.java:126)
>	at com.microsoft.playwright.impl.ChannelOwner.sendMessage(ChannelOwner.java:102)
>	at com.microsoft.playwright.impl.ChannelOwner.sendMessage(ChannelOwner.java:98)
>	at com.microsoft.playwright.impl.BrowserImpl.closeImpl(BrowserImpl.java:84)
>	at com.microsoft.playwright.impl.BrowserImpl.lambda$close$0(BrowserImpl.java:71)
>	at com.microsoft.playwright.impl.LoggingSupport.lambda$withLogging$0(LoggingSupport.java:36)
>	at com.microsoft.playwright.impl.LoggingSupport.withLogging(LoggingSupport.java:47)
>	at com.microsoft.playwright.impl.ChannelOwner.withLogging(ChannelOwner.java:87)
>	at com.microsoft.playwright.impl.LoggingSupport.withLogging(LoggingSupport.java:35)
>	at com.microsoft.playwright.impl.BrowserImpl.close(BrowserImpl.java:71)
```

Using a debugger you can see that a proper exception is thrown, but is swallowed:

```
> com.microsoft.playwright.impl.DriverException: Error {
>   message='Executable doesn't exist at %appdata%\Local\ms-playwright\ffmpeg-1008\ffmpeg-win64.exe
> ╔════════════════════════════════════════════════════════════════════════════════════════════╗
> ║ Looks like Playwright was just installed or updated.                                       ║
> ║ Please run the following command to download new browsers:                                 ║
> ║                                                                                            ║
> ║     mvn exec:java -e -D exec.mainClass=com.microsoft.playwright.CLI -D exec.args="install" ║
> ║                                                                                            ║
> ║ <3 Playwright Team                                                                         ║
> ╚════════════════════════════════════════════════════════════════════════════════════════════╝
>   name='Error
>   stack='Error: Executable doesn't exist at %appdata%\Local\ms-playwright\ffmpeg-1008\ffmpeg-win64.exe
> ╔════════════════════════════════════════════════════════════════════════════════════════════╗
> ║ Looks like Playwright was just installed or updated.                                       ║
> ║ Please run the following command to download new browsers:                                 ║
> ║                                                                                            ║
> ║     mvn exec:java -e -D exec.mainClass=com.microsoft.playwright.CLI -D exec.args="install" ║
> ║                                                                                            ║
> ║ <3 Playwright Team                                                                         ║
> ╚════════════════════════════════════════════════════════════════════════════════════════════╝
>     at executablePathOrDie (%appdata%\Local\Temp\playwright-java-2145463888745029214\package\lib\server\registry\index.js:305:15)
>     at Object.executablePathOrDie (%appdata%\Local\Temp\playwright-java-2145463888745029214\package\lib\server\registry\index.js:481:43)
>     at FrameSession._createVideoRecorder (%appdata%\Local\Temp\playwright-java-2145463888745029214\package\lib\server\chromium\crPage.js:793:68)
>     at FrameSession._initialize (%appdata%\Local\Temp\playwright-java-2145463888745029214\package\lib\server\chromium\crPage.js:410:18)
>     at async CRBrowserContext.newPage (%appdata%\Local\Temp\playwright-java-2145463888745029214\package\lib\server\browserContext.js:356:25)
>     at async BrowserContextDispatcher.newPage (%appdata%\Local\Temp\playwright-java-2145463888745029214\package\lib\server\dispatchers\browserContextDispatcher.js:182:55)
>     at async DispatcherConnection.dispatch (%appdata%\Local\Temp\playwright-java-2145463888745029214\package\lib\server\dispatchers\dispatcher.js:312:22)
> }
```

**Comment by yury-s** [comment]
Sorry this is the error I get when running in terminal, it contains java stack trace as well as instruction on how to fix it, where is the exception swallowed?

```
$ PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1 mvn -e exec:java -Dexec.mainClass=org.example.PrintTitle
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------< org.example:examples >------------------------
[INFO] Building Playwright Client Examples 1.34.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.1.0:java (default-cli) @ examples ---
Skipping browsers download because `PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD` env variable is set
/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/chromium/crPage.js:377
      this._firstNonInitialNavigationCommittedReject(new Error('Page closed'));
                                                     ^

Error: Page closed
    at CRSession.<anonymous> (/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/chromium/crPage.js:377:54)
    at Object.onceWrapper (node:events:627:28)
    at CRSession.emit (node:events:525:35)
    at /private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/chromium/crConnection.js:211:39

Node.js v18.16.0
[WARNING] 
com.microsoft.playwright.PlaywrightException: Error {
  message='Executable doesn't exist at /Users/yurys/Library/Caches/ms-playwright/ffmpeg-1009/ffmpeg-mac
╔════════════════════════════════════════════════════════════════════════════════════════════╗
║ Looks like Playwright was just installed or updated.                                       ║
║ Please run the following command to download new browsers:                                 ║
║                                                                                            ║
║     mvn exec:java -e -D exec.mainClass=com.microsoft.playwright.CLI -D exec.args="install" ║
║                                                                                            ║
║ <3 Playwright Team                                                                         ║
╚════════════════════════════════════════════════════════════════════════════════════════════╝
  name='Error
  stack='Error: Executable doesn't exist at /Users/yurys/Library/Caches/ms-playwright/ffmpeg-1009/ffmpeg-mac
╔════════════════════════════════════════════════════════════════════════════════════════════╗
║ Looks like Playwright was just installed or updated.                                       ║
║ Please run the following command to download new browsers:                                 ║
║                                                                                            ║
║     mvn exec:java -e -D exec.mainClass=com.microsoft.playwright.CLI -D exec.args="install" ║
║                                                                                            ║
║ <3 Playwright Team                                                                         ║
╚════════════════════════════════════════════════════════════════════════════════════════════╝
    at executablePathOrDie (/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/registry/index.js:305:15)
    at Object.executablePathOrDie (/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/registry/index.js:481:43)
    at FrameSession._createVideoRecorder (/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/chromium/crPage.js:792:68)
    at FrameSession._initialize (/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/chromium/crPage.js:409:18)
    at async CRBrowserContext.newPage (/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/browserContext.js:356:25)
    at async BrowserContextDispatcher.newPage (/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/dispatchers/browserContextDispatcher.js:199:55)
    at async DispatcherConnection.dispatch (/private/var/folders/t3/_mbh6z155sz1cr7prkpn9lsh0000gn/T/playwright-java-6001006418829335858/package/lib/server/dispatchers/dispatcher.js:312:22)
}
    at com.microsoft.playwright.impl.WaitableResult.get (WaitableResult.java:54)
    at com.microsoft.playwright.impl.ChannelOwner.runUntil (ChannelOwner.java:111)
    at com.microsoft.playwright.impl.Connection.sendMessage (Connection.java:126)
    at com.microsoft.playwright.impl.ChannelOwner.sendMessage (ChannelOwner.java:102)
    at com.microsoft.playwright.impl.ChannelOwner.sendMessage (ChannelOwner.java:98)
    at com.microsoft.playwright.impl.BrowserContextImpl.newPageImpl (BrowserContextImpl.java:375)
    at com.microsoft.playwright.impl.BrowserContextImpl.lambda$newPage$13 (BrowserContextImpl.java:368)
    at com.microsoft.playwright.impl.LoggingSupport.withLogging (LoggingSupport.java:47)
    at com.microsoft.playwright.impl.ChannelOwner.withLogging (ChannelOwner.java:87)
    at com.microsoft.playwright.impl.BrowserContextImpl.newPage (BrowserContextImpl.java:368)
    at com.microsoft.playwright.impl.BrowserContextImpl.newPage (BrowserContextImpl.java:44)
    at org.example.PrintTitle.main (PrintTitle.java:29)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:279)
    at java.lang.Thread.run (Thread.java:1623)
Caused by: com.microsoft.playwright.impl.DriverException: Error {
  message='Executable doesn't exist at /Users/yurys/Library/Caches/ms-playwright/ffmpeg-1009/ffmpeg-mac
...
```

**Comment by NoonRightsWarriorBehindHovering** [comment]
I'm unable to reproduce the exact Error Message from above. I narrowed down my Issue to a weird conflict between Lombok's @Cleanup Annotation and Playwright. In particular i would expect the following Code to all work equally:

With Lombok
```java
        // Does not work
        try (Playwright playwright = Playwright.create()) {
            @Cleanup Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions().setChannel("msedge"));
            BrowserContext context = browser.newContext(new Browser.NewContextOptions()
                .setRecordVideoDir(Paths.get("target", "videos")));
            Page page = context.newPage();
            page.navigate("https://example.com");
            // ...
        }
```

Which basically compiles to
```java
        // Does not work
        try (Playwright playwright = Playwright.create()) {
            Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions().setChannel("msedge"));
            try {
                BrowserContext context = browser.newContext(new Browser.NewContextOptions()
                    .setRecordVideoDir(Paths.get("target", "videos")));
                Page page = context.newPage();
                page.navigate("https://example.com");
            } finally {
                if (browser != null) browser.close();
            }
            // ...
        }
```

While using the equivalent Java 8 constructs do work:

```java
       //  Works
        try (Playwright playwright = Playwright.create()) {
            try (Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions().setChannel("msedge"))) {
                BrowserContext context = browser.newContext(new Browser.NewContextOptions()
                    .setRecordVideoDir(Paths.get("target", "videos")));
                Page page = context.newPage();
                page.navigate("https://example.com");
            }
            // ...
        }
```

Unsure why my Test Case above no longer works identically for me, but i guess that's to be expected if one does not isolate Test Cases for further studying.

```
<path>package\lib\server\chromium\crPage.js:378
      this._firstNonInitialNavigationCommittedReject(new Error('Page closed'));
                                                     ^

Error: Page closed
    at CRSession.<anonymous> (<path>package\lib\server\chromium\crPage.js:378:54)
    at Object.onceWrapper (node:events:627:28)
    at CRSession.emit (node:events:525:35)
    at <path>package\lib\server\chromium\crConnection.js:211:39

Node.js v18.16.0
[WARNING] 
com.microsoft.playwright.PlaywrightException: Failed to read message from driver, pipe closed.
    at com.microsoft.playwright.impl.PipeTransport.poll (PipeTransport.java:75)
    at com.microsoft.playwright.impl.Connection.processOneMessage (Connection.java:200)
    at com.microsoft.playwright.impl.ChannelOwner.runUntil (ChannelOwner.java:109)
    at com.microsoft.playwright.impl.Connection.sendMessage (Connection.java:126)
    at com.microsoft.playwright.impl.ChannelOwner.sendMessage (ChannelOwner.java:102)
    at com.microsoft.playwright.impl.ChannelOwner.sendMessage (ChannelOwner.java:98)
    at com.microsoft.playwright.impl.BrowserImpl.closeImpl (BrowserImpl.java:84)
    at com.microsoft.playwright.impl.BrowserImpl.lambda$close$0 (BrowserImpl.java:71)
    at com.microsoft.playwright.impl.LoggingSupport.lambda$withLogging$0 (LoggingSupport.java:36)
    at com.microsoft.playwright.impl.LoggingSupport.withLogging (LoggingSupport.java:47)
    at com.microsoft.playwright.impl.ChannelOwner.withLogging (ChannelOwner.java:87)
    at com.microsoft.playwright.impl.LoggingSupport.withLogging (LoggingSupport.java:35)
    at com.microsoft.playwright.impl.BrowserImpl.close (BrowserImpl.java:71)
    at org.example.ExampleReproducible.main (ExampleReproducible.java:31)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:279)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.io.EOFException
    at com.microsoft.playwright.impl.ReaderThread.readIntLE (PipeTransport.java:110)
    at com.microsoft.playwright.impl.ReaderThread.readMessage (PipeTransport.java:139)
    at com.microsoft.playwright.impl.ReaderThread.run (PipeTransport.java:125)
```

**Comment by yury-s** [comment]
> With Lombok

Looking at the description of the project I would not be surprised if it tampers with the playwright exceptions.

Here is the repro project I use: https://github.com/yury-s/java-issue-1280. If you deem this is a Playwright bug, please provide a complete self-contained repro like that with vanilla Playwright. Make sure you run it against latest Playwright version.

**Comment by NoonRightsWarriorBehindHovering** [comment]
> If you deem this is a Playwright bug, please provide a complete self-contained repro like that with vanilla Playwright. Make sure you run it against latest Playwright version.

Quite Simple to add it to your Repository. I've forked it here using a new Code Space. It uses a newer Version of maven than you did in your example output. 

Just for completeness sake, I'll include the output of mvn -v :
Maven Version:
    Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
    Maven home: /usr/local/sdkman/candidates/maven/current
    Java version: 11.0.14.1, vendor: Microsoft, runtime: /opt/java/11.0.14
    Default locale: en, platform encoding: UTF-8
    OS name: "linux", version: "5.4.0-1106-azure", arch: "amd64", family: "unix"

I did not change the Version of Playwright, the SHA-1 is:
435c8720277acc9a4640fb91d67b3601f1b94d27 and only Version 1.33 is installed on the codespace.

As i easily reproduced that on Windows too, i don't think this is a "Lombok" or "Environment" bug, but rather that closing the Browser environment in a finally block is unsupported. Which is okay-ish, as that is easily worked around (see my original Example) with try-with-resources.

For readability purposes i do prefer using @Cleanup though and using try-with-resources with the browser object directly is not always an option.

For the fork see: https://github.com/NoonRightsWarriorBehindHovering/mre-playwright-1
I renamed it, to fit with my other repository naming conventions.

Just to be clear again: It is definitely not a "lombok" issue, but a playwright issue. Alternatively it is a "Java" issue, but i don't believe that to be the case.
